contrib/confluentinc/confluent-kafka-go/kafka/option.go

```diff
@@ -42,7 +49,8 @@ func WithContext(ctx context.Context) Option {
 // WithServiceName sets the config service name to serviceName.
 func WithServiceName(serviceName string) Option {
 	return func(cfg *config) {
-		cfg.serviceName = serviceName
+		cfg.consumerServiceName = serviceName
+		cfg.producerServiceName = serviceName
```

Comment:
I get that this preserves the existing behavior (when there was just `cfg.serviceName`). But if someone wants to set a specific service for their producer spans, how would they do so? Do we offer that functionality?

Reply:
No, we don't support using different service names for clients/producers for the same client. It's assumed you will use one client for producer and another for consumer. They can be configured separately and given different service names.

If that's something we want we can open a ticket for it.